### PR TITLE
build: remove CircleCI

### PR DESCRIPTION
**Pull request by VerteDinde**
#### Description of Change

🫡 This PR removes CircleCI from e/e CI runs.

#### Checklist

- [ ] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [ ] relevant documentation, tutorials, templates and examples are changed or added
- [ ] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: none

### CODE REVIEW

File: spec/api-app-spec.ts
Comment (jkleinsc) [inline spec/api-app-spec.ts:1563]: This whole comment should be removed (we don't use VSTS anymore).

**Review by jkleinsc — CHANGES_REQUESTED** [review]
Just a small cleanup is needed on the spec comment.

**Comment by jkleinsc** [comment]
- Do not merge until #42722 is merged.

**Review by MarshallOfSound — COMMENTED** [review]
Few places we ref "circle" still

* https://github.com/electron/electron/blob/1e1dc22e167a9b7313aaa914fb400a19edbbeb7f/spec/lib/codesign-helpers.ts#L22
* https://github.com/electron/electron/blob/1e1dc22e167a9b7313aaa914fb400a19edbbeb7f/script/push-patch.js#L4
* https://github.com/electron/electron/blob/1e1dc22e167a9b7313aaa914fb400a19edbbeb7f/.env.example#L5
* https://github.com/electron/electron/blob/1e1dc22e167a9b7313aaa914fb400a19edbbeb7f/spec/version-bump-spec.ts#L93
* https://github.com/electron/electron/blob/1e1dc22e167a9b7313aaa914fb400a19edbbeb7f/.devcontainer/devcontainer.json#L39

**Review by jkleinsc — APPROVED** [review]

**Pull request merged**
